clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -87,13 +90,36 @@
                                                                         "Metadata for each partition of the topic."));
 
     public static final Schema METADATA_RESPONSE_V0 = new Schema(new Field("brokers",
-                                                                           new ArrayOf(BROKER),
+                                                                           new ArrayOf(BROKER_V0),
                                                                            "Host and port information for all brokers."),
                                                                  new Field("topic_metadata",
                                                                            new ArrayOf(TOPIC_METADATA_V0)));
 
-    public static final Schema[] METADATA_REQUEST = new Schema[] {METADATA_REQUEST_V0};
-    public static final Schema[] METADATA_RESPONSE = new Schema[] {METADATA_RESPONSE_V0};
+    public static final Schema BROKER_V1 = new Schema(new Field("node_id", INT32, "The broker id."),
```

Comment:
Yes, my understanding is these "sub-schemas" are supposed to tie directly to single api/protocol to allow them to change independently. I think the fact that a BROKER_V0 was shared was a mistake. I had left the old one being shared just to minimize change. I can break-out and rename a bit to prevent an accidental sharing in the future.